pkg/image/registry/image/strategy.go

```diff
@@ -38,8 +38,8 @@ func (s imageStrategy) PrepareForCreate(obj runtime.Object) {
 	if err := api.ImageWithMetadata(newImage); err != nil {
 		utilruntime.HandleError(fmt.Errorf("Unable to update image metadata for %q: %v", newImage.Name, err))
 	}
-
-	s.clearSignatureDetails(newImage)
+	// allow to set the signatures only through signatures subresource
```

Comment:
Repasting @liggitt's earlier comment here:

> I'm not sure whether I'd expect the first person to push an image to be able to surface their signatures into any other namespace they want. @smarterclayton, was that what you expected?

And re-pasting my earlier comment as well:

> As discussed before, we don't want to allow to set signatures during image
push nor image import.
>
> The image rest storage is being used in other virtual rest stores (ImageStreamMapping, ImageStreamImport). It seems reasonable to me to allow admin to create/update image directly with any signatures he wants using the regular store. The other virtual rests should be passed some limited image rest though.
>
> So instead of adding InternalREST, would it be okay to:
>
>  1. add something like UnsignedREST
>  2. make the image REST fully capable